File: java_solutions/1471.the-k-strongest-values-in-an-array.java
/*
 * @lc app=leetcode id=1471 lang=java
 *
 * [1471] The k Strongest Values in an Array
 */

class Solution {
    public int[] getStrongest(int[] arr, int k) {
        return java.util.stream.Stream.<int[]>of(java.util.Arrays.stream(arr).sorted().toArray()).map(a -> java.util.Arrays.stream(a).boxed().sorted((x, y) -> Math.abs(x - a[(a.length - 1) / 2]) == Math.abs(y - a[(a.length - 1) / 2]) ? y.compareTo(x) : Integer.compare(Math.abs(y - a[(a.length - 1) / 2]), Math.abs(x - a[(a.length - 1) / 2]))).limit(k).mapToInt(Integer::intValue).toArray()).findFirst().get();
    }
}

File: kotlin_solutions_1/1471.the-k-strongest-values-in-an-array.kt
/*
 * @lc app=leetcode id=1471 lang=kotlin
 *
 * [1471] The k Strongest Values in an Array
 */

class Solution {
    fun getStrongest(arr: IntArray, k: Int): IntArray = arr.sorted().let { sorted -> sorted[(sorted.size - 1) / 2].let { m -> arr.sortedDescending().sortedByDescending { Math.abs(it - m) }.take(k).toIntArray() } }
}